source/extensions/transport_sockets/proxy_protocol/proxy_protocol.cc

```diff
@@ -165,6 +169,49 @@ void UpstreamProxyProtocolSocketFactory::hashKey(
   }
 }
 
+std::vector<Envoy::Network::ProxyProtocolTLV> UpstreamProxyProtocolSocket::buildCustomTLVs() {
+  std::vector<Envoy::Network::ProxyProtocolTLV> custom_tlvs;
+
```

Comment:
I think you should support setting the TLVs in the transport socket config, in addition to host metadata, for cases where you want to send the same values for every host, or a default set of values with only some hosts overriding.

Reply:
Agreed. I'll look into adding this locally and adding tests that exercise the new precedence chain between custom TLVs defined in host metadata, the transport socket config, etc.

Reply:
Added this to the ProxyProtocolConfig protobuf message in https://github.com/envoyproxy/envoy/pull/37591/commits/d0a3b622e5727e52cee9117de5dba73b8a57419b. I went back-and-forth on whether this belonged on the ProxyProtocolUpstreamTransport message instead, but the former allowed for an easy implementation. Let me know if you have thoughts on whether that's the right home for this.